diff --git a/contrib/ParseAndAddCatchTests.cmake b/contrib/ParseAndAddCatchTests.cmake
index 5e89cb76..b301f0f6 100644
--- a/contrib/ParseAndAddCatchTests.cmake
+++ b/contrib/ParseAndAddCatchTests.cmake
@@ -24,7 +24,9 @@
 #        add_executable(${PROJECT_NAME} ${SOURCE_FILES})                                           #
 #                                                                                                  #
 #        include(ParseAndAddCatchTests)                                                            #
-#        ParseAndAddCatchTests(${PROJECT_NAME})                                                    #
+#        ParseAndAddCatchTests(${PROJECT_NAME}                                                     #
+#                              [TargetVar <targetVar>]                                             #
+#                              [ExtraLabels <extraLabels>])                                        #
 #                                                                                                  #
 # The following variables affect the behavior of the script:                                       #
 #                                                                                                  #
@@ -93,6 +95,7 @@ function(ParseAndAddCatchTests_ParseFile SourceFile TestTarget)
         ParseAndAddCatchTests_PrintDebugMessage("Detected OBJECT library: ${SourceFile} this will not be scanned for tests.")
         return()
     endif()
+    cmake_parse_arguments(PF "" "TargetsVar" "ExtraLabels" ${ARGN})
     # According to CMake docs EXISTS behavior is well-defined only for full paths.
     get_filename_component(SourceFile ${SourceFile} ABSOLUTE)
     if(NOT EXISTS ${SourceFile})
@@ -168,7 +171,7 @@ function(ParseAndAddCatchTests_ParseFile SourceFile TestTarget)
           unset(Tags)
         endif()
 
-        list(APPEND Labels ${Tags})
+        list(APPEND Labels ${Tags} ${PF_ExtraLabels})
 
         set(HiddenTagFound OFF)
         foreach(label ${Labels})
@@ -196,6 +199,9 @@ function(ParseAndAddCatchTests_ParseFile SourceFile TestTarget)
             endif()
             # Add the test and set its properties
             add_test(NAME "${CTestName}" COMMAND ${OptionalCatchTestLauncher} $<TARGET_FILE:${TestTarget}> ${Name} ${AdditionalCatchParameters})
+            if(PF_TargetsVar)
+              list(APPEND ${PF_TargetsVar} "${CTestName}")
+            endif()
             # Old CMake versions do not document VERSION_GREATER_EQUAL, so we use VERSION_GREATER with 3.8 instead
             if(PARSE_CATCH_TESTS_NO_HIDDEN_TESTS AND ${HiddenTagFound} AND ${CMAKE_VERSION} VERSION_GREATER "3.8")
                 ParseAndAddCatchTests_PrintDebugMessage("Setting DISABLED test property")
@@ -216,15 +222,23 @@ function(ParseAndAddCatchTests_ParseFile SourceFile TestTarget)
 
 
     endforeach()
+
+    if(PF_TargetsVar)
+      set(${PF_TargetsVar} "${${PF_TargetsVar}}" PARENT_SCOPE)
+    endif()
 endfunction()
 
 # entry point
 function(ParseAndAddCatchTests TestTarget)
+    cmake_parse_arguments(PACT "" "TargetsVar" "ExtraLabels" ${ARGN})
     ParseAndAddCatchTests_PrintDebugMessage("Started parsing ${TestTarget}")
     get_target_property(SourceFiles ${TestTarget} SOURCES)
     ParseAndAddCatchTests_PrintDebugMessage("Found the following sources: ${SourceFiles}")
     foreach(SourceFile ${SourceFiles})
-        ParseAndAddCatchTests_ParseFile(${SourceFile} ${TestTarget})
+        ParseAndAddCatchTests_ParseFile(${SourceFile} ${TestTarget} ${ARGN})
     endforeach()
     ParseAndAddCatchTests_PrintDebugMessage("Finished parsing ${TestTarget}")
+    if(PACT_TargetsVar)
+      set(${PACT_TargetsVar} "${${PACT_TargetsVar}}" PARENT_SCOPE)
+    endif()
 endfunction()
